library/kani_macros/src/sysroot/contracts/shared.rs

```diff
@@ -53,7 +53,34 @@ impl<'a> ContractConditionsHandler<'a> {
                 #[allow(dead_code, unused_variables, unused_mut)]
             ));
         }
+
+        #[cfg(not(feature = "no_core"))]
         self.output.extend(self.annotated_fn.attrs.iter().flat_map(Attribute::to_token_stream));
+
+        // When verifying core and standard library, we need to add an unstable attribute to
+        // the functions generated by Kani.
+        // We also need to filter `rustc_diagnostic_item` attribute.
+        // We should consider a better strategy than just duplicating all attributes.
+        #[cfg(feature = "no_core")]
+        {
+            self.output.extend(quote!(
+                #[unstable(feature="kani", issue="none")]
+            ));
+            self.output.extend(
+                self.annotated_fn
+                    .attrs
+                    .iter()
+                    .filter(|attr| {
+                        if let Some(ident) = attr.path().get_ident() {
+                            let name = ident.to_string();
+                            !name.starts_with("rustc") && !(name == "stable")
```

Comment:
Is there any chance of an attribute that is not this, but shares these properties?  I suppose we can always deal with that case later.

Reply:
Yes. For sure there is! That's why I mentioned that we need to find a more robust way to do this. I was thinking about creating an associated function instead. @JustusAdam do you remember why we decided against it?